homeassistant/components/geo_json_events/geo_location.py

```diff
@@ -59,63 +60,72 @@ def setup_platform(
     )
     radius_in_km = config[CONF_RADIUS]
     # Initialize the entity manager.
-    feed = GeoJsonFeedEntityManager(
-        hass, add_entities, scan_interval, coordinates, url, radius_in_km
+    manager = GeoJsonFeedEntityManager(
+        hass, async_add_entities, scan_interval, coordinates, url, radius_in_km
     )
+    await manager.async_init()
 
-    def start_feed_manager(event):
+    @callback
+    async def start_feed_manager(event=None):
         """Start feed manager."""
-        feed.startup()
+        await manager.async_update()
 
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_feed_manager)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_feed_manager)
 
 
 class GeoJsonFeedEntityManager:
     """Feed Entity Manager for GeoJSON feeds."""
 
     def __init__(
-        self, hass, add_entities, scan_interval, coordinates, url, radius_in_km
+        self, hass, async_add_entities, scan_interval, coordinates, url, radius_in_km
     ):
         """Initialize the GeoJSON Feed Manager."""
 
         self._hass = hass
+        websession = aiohttp_client.async_get_clientsession(hass)
         self._feed_manager = GenericFeedManager(
+            websession,
             self._generate_entity,
             self._update_entity,
             self._remove_entity,
             coordinates,
             url,
             filter_radius=radius_in_km,
         )
-        self._add_entities = add_entities
+        self._async_add_entities = async_add_entities
         self._scan_interval = scan_interval
 
-    def startup(self):
-        """Start up this manager."""
-        self._feed_manager.update()
-        self._init_regular_updates()
+    async def async_init(self):
+        """Schedule initial and regular updates based on configured time interval."""
 
-    def _init_regular_updates(self):
-        """Schedule regular updates at the specified interval."""
-        track_time_interval(
-            self._hass, lambda now: self._feed_manager.update(), self._scan_interval
-        )
+        async def update(event_time):
+            """Update."""
+            await self.async_update()
+
+        # Trigger updates at regular intervals.
+        async_track_time_interval(self._hass, update, self._scan_interval)
```

Comment:
A future improvement would be to use our coordinator update helper instead.

Reply:
Yes, agree. I wanted to keep this PR light-weight and migrate to the coordinator later.